sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_authentication.py

```diff
@@ -62,8 +67,84 @@ def on_challenge(self, request, response):  # pylint:disable=unused-argument
         return False
 
 
-def _parse_claims_challenge(challenge):
-    # type: (str) -> Optional[str]
+# pylint:disable=too-few-public-methods
+class _AuxiliaryAuthenticationPolicyBase:
+    """Adds auxiliary authorization token header to requests.
+
+    :param ~azure.core.credentials.TokenCredential auxiliary_credentials: auxiliary credential for authorizing requests
+    :param str scopes: required authentication scopes
+    """
+
+    def __init__(
+            self, auxiliary_credentials, *scopes, **kwargs  # pylint: disable=unused-argument
+    ):
+        self._auxiliary_credentials = auxiliary_credentials
+        self._scopes = scopes
+        self._aux_tokens = None
+
+    @staticmethod
```

Comment:
We generally don't use static methods. But since this is a private API, it is fine. I assume this was only done to avoid a warning about unused `self`?

Reply:
This is my understanding too. :)